src/rgw/rgw_rados.h

```diff
@@ -1270,6 +1272,11 @@ struct RGWZoneParams : RGWSystemMetaObj {
     } else {
       reshard_pool = name + ".rgw.reshard";
     }
+    if (struct_v >= 11) {
+      ::decode(groups_pool, bl);
+    } else {
+      roles_pool = name + ".rgw.groups";
```

Comment:
@cbodley : If the version is older than 11, shouldnt we fallback to the old style of naming pools? group_name is set to .rgw.meta:groups in RGWZoneParams::fix_pool_names(). roles_pool is a typo here which I will correct.